lib/SQLBuilder.php

```diff
@@ -113,36 +113,30 @@ public function get_where_values(): array
         return $this->where_values;
     }
 
-    public function where(/* (conditions, values) || (hash) */): static
+    /**
+     * @param array<WhereClause>   $clauses
+     * @param array<string,string> $mappedNames
+     *
+     * @throws Exception\ExpressionsException
+     *
+     * @return $this
+     */
+    public function where(array $clauses=[], array $mappedNames=[]): static
     {
-        $args = func_get_args();
-        $num_args = count($args);
-
-        if (1 == $num_args && is_hash($args[0])) {
-            $hash = empty($this->joins) ? $args[0] : $this->prepend_table_name_to_fields($args[0]);
-            $e = new Expressions($this->connection, $hash);
-            $this->where = $e->to_s();
-            $this->where_values = array_flatten($e->values());
-        } elseif ($num_args > 0) {
-            // if the values has a nested array then we'll need to use Expressions to expand the bind marker for us
-            $values = array_slice($args, 1);
-
-            foreach ($values as $name => &$value) {
-                if (is_array($value)) {
-                    $e = new Expressions($this->connection, $args[0]);
-                    $e->bind_values($values);
-                    $this->where = $e->to_s();
-                    $this->where_values = array_flatten($e->values());
-
-                    return $this;
-                }
-            }
-
-            // no nested array so nothing special to do
-            $this->where = $args[0] ?? '';
-            $this->where_values = $values;
+        $values = [];
+        $sql = '';
+        $glue = ' AND ';
+        foreach ($clauses as $idx => $clause) {
+            $expression = $clause->to_s($this->connection, !empty($this->joins) ? $this->table : '', $mappedNames);
+            $values = array_merge($values, array_flatten($clause->values()));
+            $inverse = $clause->inverse() ? '!' : '';
```

Comment:
Only now do I know what inverse() does. Rename to negate() or toBeNegated()

Reply:
I'll meet you in the middle and go with `negated`. 46a545f